src/Controller/Component/AclComponent.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Tools\Controller\Component;
+
+use Shim\Controller\Component\Component;
+use Cake\Controller\ComponentRegistry;
+use Cake\Network\Request;
+use Cake\Auth;
+
+/**
+ * A class for checking access permissions.
+ *
+ * @author Bernhard Picher
+ * @license MIT
+ */
+class AclComponent extends Component {
+
+	public $components = ['Auth'];
+
+	/**
+	 * AclComponent:check ( $controller, $action, $plugin )
+	 * 
+	 * Checks if the currently logged in user would have access to the given request.
+	 *
+	 * @param string $controller Controller name.
+	 * @param string $action Action name.
+	 * @param string $plugin Plugin name. Defaults to `null`.
+     * @return bool True if access would be granted
+	 */
+	public function check ( $controller, $action, $plugin = 'null' ) {
```

Comment:
Good question. At the time it was clear that in 3.x there is the possibility now of having the same controller in multiple plugins because of namespacing. Not sure if this is implemented just yet.